src/core/services/compiler/compiler.js

```diff
@@ -248,13 +266,8 @@ MdCompilerService.prototype._compileElement = function(locals, element, options)
         $element: element
       });
 
-      var invokeCtrl = self.$controller(options.controller, injectLocals, true, options.controllerAs);
-
-      if (options.bindToController) {
-        angular.extend(invokeCtrl.instance, locals);
-      }
-
-      var ctrl = invokeCtrl();
+      // Create the specified controller instance.
+      var ctrl = self._createController(options, injectLocals);
```

Comment:
Not directly related to this PR, but it seems that you are accidentally putting `$element` into `locals`. Essentially, `locals` and `injectLocals` are the same object, but this doesn't seem intentional (and indeed assigning `$element` to the controller instance seems strange).

Was it intentional?